web-app/src/lib/job-input/job-input.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from "zod";
+
+import { JobInputSchemaIntlPath, ValidJobInputTypes } from "./type";
+import { isTypeAndValidationValid } from "./util";
+import { validationSchema } from "./validation";
+
+export const jobInputSchema = (t?: IntlTranslation<JobInputSchemaIntlPath>) =>
+  z
+    .object({
+      id: z.string().nonempty({
+        message: t?.("Id.required"),
+      }),
+      type: z.enum(ValidJobInputTypes, {
+        message: t?.("Type.enum", {
+          options: ValidJobInputTypes.join(", "),
+        }),
+      }),
+      name: z.string().nonempty({
+        message: t?.("Name.required"),
+      }),
+      data: z.object({
+        values: z
```

Comment:
This should only be allowed if the type is option

to achieve that you can have two complete zod objects including name id… connected with or (one for option, one for all other, maybe have other groups for type none)

Reply:
@schaier-io did you look at lines 50-65 of the same file?